toolchain/lex/tokenized_buffer_benchmark.cpp

```diff
@@ -427,6 +427,66 @@ void BM_ValidKeywords(benchmark::State& state) {
 }
 BENCHMARK(BM_ValidKeywords);
 
+void BM_ValidKeywordsAsRawIdentifiers(benchmark::State& state) {
+  llvm::SmallVector<std::string> raw_keywords;
+  raw_keywords.resize(TokenKind::KeywordTokens.size(), "r#");
+  for (auto [keyword, raw_keyword] :
+       llvm::zip(TokenKind::KeywordTokens, raw_keywords)) {
+    raw_keyword.append(keyword.fixed_spelling());
+  }
+
+  absl::BitGen gen;
+  std::array<llvm::StringRef, NumTokens> tokens;
+  for (int i : llvm::seq(NumTokens)) {
+    tokens[i] = raw_keywords[i % raw_keywords.size()];
+  }
+  std::shuffle(tokens.begin(), tokens.end(), gen);
+  std::string source = llvm::join(tokens, " ");
+
+  LexerBenchHelper helper(source);
+  for (auto _ : state) {
+    TokenizedBuffer buffer = helper.Lex();
+    CARBON_CHECK(!buffer.has_errors());
+  }
+
+  state.SetBytesProcessed(state.iterations() * source.size());
+  state.counters["tokens_per_second"] = benchmark::Counter(
+      NumTokens, benchmark::Counter::kIsIterationInvariantRate);
+}
+BENCHMARK(BM_ValidKeywordsAsRawIdentifiers);
+
+// This benchmark does a 50-50 split of r-prefixed and r#-prefixed identifiers
```

Comment:
Is this comparing `r#`-prefixed and `r`-prefixed identifiers, or `r#`-prefixed identifiers and general non-raw identifiers?

Reply:
`r`-prefixed, not general. Note, it looks like when you reviewed you were looking at the "Comments" commit but not the "bm" commit; the "Comments" commit did general.

https://github.com/carbon-language/carbon-lang/pull/3344/commits/2f5d94374e20d98c330bd074af4174bf214856ef